helper/src/cast.rs

```diff
@@ -3,49 +3,95 @@
 //! This modules provides utilities for casting between types.
 //!
 //! `#[no_std]` compatible.
+//!
+//! # 64-bit invariant
+//! This module is available on 32-bit arches although panics
+//! will occur between lossy casts, e.g. [`u64_to_usize`] where
+//! the input is larger than [`u32::MAX`].
+//!
+//! On 64-bit arches, all functions are lossless.
+
+// TODO:
+// These casting functions are heavily used throughout the codebase
+// yet it is not enforced that all usages are correct in 32-bit cases.
+// Panicking may be a short-term solution - find a better fix for 32-bit arches.
```

Comment:
The issue for `wasm32-unknown-unknown` builds was:
- `cuprate_rpc_types` depends on `cuprate_types` with `json` feature
- `json` feature requires `cuprate_helper::cast`

To fix this, I am making `cuprate_helper::cast` work on 32-bit targets, yet panic when the cast is lossy. I don't think this is the correct long-term solution but it does work for now.